controllers/storageclient_controller.go

```diff
@@ -210,9 +217,25 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "Secret":
+			if eResource.Name == fmt.Sprintf("noobaa-remote-join-secret-%s", r.storageClient.Status.ConsumerID) {
+				data := map[string]string{}
+				if err := json.Unmarshal(eResource.Data, &data); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshall noobaa join secret data: %v", err)
+				}
+
+				if err := r.reconcileNoobaaJoinSecret(data); err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+		}
+	}
+	// reconcile  remote noobaa only if it is not a local client
+	if _, exists := r.storageClient.GetAnnotations()["is-local-client"]; !exists {
+		if err := r.reconcileRemoteNoobaa(); err != nil {
+			return reconcile.Result{}, err
```

Comment:
1. who will put the annotation on the client?
2. Doesn't noobaa reconciliation here depend on the existing of the secret? Aren't they coupled and if so why is this not dependent on the existence of the secret instead of an annotation?

Reply:
1. ocs storageclient controller is going to put this annotation on the local client.
ref: https://github.com/red-hat-storage/ocs-operator/pull/2680/files#diff-3a3043bcbec115677bf1cd899c5c562a02eb74b2c874f80b42ff8bd6880c41f4R43
2. Yes but ocs-client needs to know if it will create the client noobaa CR, for provider clusters it will not create the same. Also local ocs-client will not have a join secret

Reply:
As discussed, not depending on the annotation, only existence of join secret